### ISSUE TITLE
can't share files

### ISSUE THREAD

**Issue by danielgomon** [emesene/emesene#808]
can't share files with anyone, I'm in emesene 2.11.7 in Arch. Here is the output of the terminal, I'm trying to share a file with a contact who uses emesene 2.11.7 in Ubuntu Lucid, the contact see the notification with notify-osd but the emesene window doesn't show the buttom to accept it...

$ emesene
*\* Message: pygobject_register_sinkfunc is deprecated (GstObject)
[12:37:42 ERROR root] Could not find any typelib for Unity
[12:37:42 ERROR root] Could not find any typelib for Dbusmenu
[12:37:43 ERROR papylib.Events] You need gstreamer to use the Audio/Video calls support
Traceback (most recent call last):
  File "/usr/share/emesene/e3/papylib/Worker.py", line 70, in <module>
    import PapyConference
  File "/usr/share/emesene/e3/papylib/PapyConference.py", line 27, in <module>
    import farsight
ImportError: No module named farsight
[12:37:59 WARNING papyon.service] Metadata: Invalid metadata
[12:37:59 WARNING papyon.service] Metadata: Invalid metadata

*\* (emesene:4263): WARNING **: Error loading plugin: libhspell.so.0: no se puede abrir el fichero del objeto compartido: No existe el fichero o el directorio

warning: line 6172: incompatible stripping characters and condition
warning: line 6184: incompatible stripping characters and condition
[12:38:00 WARNING papyon.p2p] Trying to publish an empty MSNObject
[12:38:00 WARNING papyon.p2p] Trying to publish an empty MSNObject
/usr/share/emesene/gui/gtkui/FileTransferWidget.py:69: GtkWarning: gtk_box_pack: assertion `child->parent == NULL' failed
  self.pack_start(self.menu)

**Comment by c10ud** [comment]
install python-farsight and some dbus stuff, seems like you have some weird installation with missing libraries and stuff

**Comment by c10ud** [comment]
(python-dbus)

**Comment by danielgomon** [comment]
thanks c10ud... it seems like there is no python-fairsight package in Arch repos (only fairsight2), I already have dbus-python installed and also have to install hspell (according with the output in the terminal)... 

Heres is the output of the package search in repos...

$ yaourt -s dbus
1 core/dbus-core 1.4.14-1 [installed]
    Freedesktop.org message bus system
2 extra/dbus 1.4.14-1 [installed]
    Freedesktop.org message bus system
3 extra/dbus-glib 0.94-2 [installed]
    GLib bindings for DBUS
4 extra/dbus-python 0.84.0-1 [installed]
    Python bindings for DBUS

$ yaourt -s farsight
1 extra/farsight2 0.0.29-1 [installed]
    Audio/Video conference software for Instant Messengers
2 extra/telepathy-farsight 0.0.19-1
    A telepathy-backend to use stream engine.

Still not working, I can't share files... here is the new output...

$ emesene
*\* Message: pygobject_register_sinkfunc is deprecated (GstObject)
[14:04:56 ERROR root] Could not find any typelib for Unity
[14:04:56 ERROR root] Could not find any typelib for Dbusmenu
warning: line 6172: incompatible stripping characters and condition
warning: line 6184: incompatible stripping characters and condition
[14:05:19 WARNING papyon.service] Metadata: Invalid metadata
[14:05:19 WARNING papyon.service] Metadata: Invalid metadata
[14:05:20 WARNING papyon.p2p] Trying to publish an empty MSNObject
[14:05:20 WARNING papyon.p2p] Trying to publish an empty MSNObject
/usr/share/emesene/gui/gtkui/FileTransferWidget.py:69: GtkWarning: gtk_box_pack: assertion `child->parent == NULL' failed
  self.pack_start(self.menu)

**Comment by Mystro256** [comment]
dbus-python is a requirement for emesene in Arch, although farsight isn't

If the farsight a requirement, I can make a bug report right away :)

Here's the package page if anyone wants to a take a quick look: http://www.archlinux.org/packages/community/i686/emesene/

**Comment by c10ud** [comment]
well, usually the package manager takes care of installing the
required deps.. the only remaining thing that could have issues is an
old version of papyon -- other than that i don't know what could be
possibly wrong -- i think other archers use emesene, do they have the
same problems?

2011/8/28 Mystro256
reply@reply.github.com:

**Comment by arielj** [comment]
@danielgomon, can you check if it's still present in latest git?

**Comment by Mystro256** [comment]
Well I use arch and never had an issue with sending/receiving files, I just did a test a few minutes ago and works great. I tested two emesene 2.11.7's (both arch), along with emesene 2.11.7 (arch) and pidgin (Ubuntu), and one more test with Emesene 2.11.7 (arch) and the latest Live Messenger (windows 7).
Also Papyon is definitely up to date: http://www.archlinux.org/packages/extra/any/papyon/

Is this related at all to #730? If receiving a file makes a sound, it could be just the same issue, different method.

**Comment by Mystro256** [comment]
Also, I noticed another bug, I get two notifications of "File transfer completed!" when emesene sends a file to someone else.

**Comment by danielgomon** [comment]
I installed the git version from here: https://aur.archlinux.org/packages.php?ID=28048

No luck, same problem...

$ emesene2
*\* Message: pygobject_register_sinkfunc is deprecated (GstObject)
[22:37:47 ERROR papyon.protocol.notification] Notification got error : 913 10
[22:37:50 WARNING papyon.service] Metadata: Invalid metadata
[22:37:51 WARNING papyon.p2p] Trying to publish an empty MSNObject

warning: line 6172: incompatible stripping characters and condition
warning: line 6184: incompatible stripping characters and condition
[22:38:11 WARNING papyon.p2p] Trying to publish an empty MSNObject

**Comment by c10ud** [comment]
913

Not allowed when hidden Sent in reply to commands that are not allowed
in the HDN status. The only known command that is prohibited when
hidden (before MSNP13) is XFR.

also you have a few of weird errors, i guess related to python or some
other unstable library e.g. "  incompatible stripping characters and
condition"

plus we were talking about papyon-git, is that inside emesene-git?
we'd like to keep issues related to packages/distros out of this
tracker

2011/8/30 danielgomon
reply@reply.github.com:

**Comment by Mystro256** [comment]
Emesene-git is an AUR package that just grabs the latest git and installs it into the /usr/share folders.
Also there's a small typo in the launcher for that package, I would assume that would have something to do with it. I asked the maintainer to change it but generally the AUR is use at your own risk anyway.

@danielgomon can you edit the "/usr/bin/emesene2.run" file and remove the "<b>.py</b>" at the end of the third line and see if that reduces some of your errors?

**Comment by danielgomon** [comment]
I cloned the git version following these instructions... http://blog.emesene.org/2010/10/instructions-for-testers.html

Now I can share files (only sometimes, most of the times not) but i can't receive... the 'accept transfer' buttom doesn't show (only can see it with the notify-osd pop-up message) 

Here's the output...

$ python2 ./emesene
*\* Message: pygobject_register_sinkfunc is deprecated (GstObject)
[20:06:10 ERROR papyon.protocol.notification] Notification got error : 913 10
[20:06:16 ERROR papyon.protocol.notification] Notification got error : 913 11
warning: line 6172: incompatible stripping characters and condition
warning: line 6184: incompatible stripping characters and condition
[20:06:19 WARNING papyon.service] Metadata: Invalid metadata
[20:08:09 WARNING papyon.msnp2p.session_manager] Received initial blob with SessionID=0 and non INVITE SLP data
[20:08:09 WARNING papyon.msnp2p.session_manager] Received initial blob with SessionID=0 and non INVITE SLP data
[20:08:09 WARNING papyon.msnp2p.session_manager] Received initial blob with SessionID=0 and non INVITE SLP data
[20:08:10 WARNING papyon.msnp2p.session_manager] Received initial blob with SessionID=0 and non INVITE SLP data

**Comment by Mystro256** [comment]
And you're sure your not set to invisible right?

**Comment by danielgomon** [comment]
Yes i'm sure... actually i was 'away'...

**Comment by Mystro256** [comment]
Hmm okay, just making sure ;)

Did you try the latest git of papyon too?

If not, you can install it via AUR: http://aur.archlinux.org/packages.php?ID=26472

**Comment by c10ud** [comment]
[20:06:10 ERROR papyon.protocol.notification] Notification got error : 913 10
[20:06:16 ERROR papyon.protocol.notification] Notification got error : 913 11

says hidden

**Comment by Sbte** [comment]
Maybe the other person was invisible?

**Comment by ErunamoJAZZ** [comment]
I have the same problem as danielgomon.

I leave you a pastebin with ./emesene -vvv   ==>  http://pastebin.com/UiYDJVWh

Also one with the regular session:

```
$ ./emesene 
/home/erunamo/Público/emesene/emesene/e3/jabber/Worker.py:27: DeprecationWarning: the sha module is deprecated; use the hashlib module instead
  import sha
/usr/lib/python2.6/dist-packages/Crypto/Util/randpool.py:40: RandomPool_DeprecationWarning: This application uses RandomPool, which is BROKEN in older releases.  See http://www.pycrypto.org/randpool-broken
  RandomPool_DeprecationWarning)
/home/erunamo/Público/emesene/emesene/gui/gtkui/__init__.py:88: Warning: g_object_set_qdata: assertion `G_IS_OBJECT (object)' failed
  gtk.main()
[00:20:36 WARNING papyon.p2p] Unknown MSN object, another end point might have it
[00:20:36 ERROR papyon.msnp2p.session_manager] No handler could handle euf-guid {A4268EEC-FEC5-49E5-95C3-F126696BDBF6}
[00:20:36 WARNING papyon.p2p] Unknown MSN object, another end point might have it
[00:20:36 ERROR papyon.msnp2p.session_manager] No handler could handle euf-guid {A4268EEC-FEC5-49E5-95C3-F126696BDBF6}
[00:20:36 WARNING papyon.p2p] Unknown MSN object, another end point might have it
[00:20:36 ERROR papyon.msnp2p.session_manager] No handler could handle euf-guid {A4268EEC-FEC5-49E5-95C3-F126696BDBF6}
[00:20:37 ERROR papyon.gnet.HTTP] Received error code 404 (Not Found) from blufiles.storage.msn.com:80
[00:20:38 ERROR papyon.gnet.HTTP] Received error code 404 (Not Found) from blufiles.storage.msn.com:80
[00:20:38 ERROR papyon.service.content_roaming] Content Roaming service got error: HTTP Error (404): Not Found
[00:20:45 WARNING papyon.p2p] Deprecated: Using MSNObject creator to guess peer
[00:20:46 WARNING papyon.msnp2p.session_manager] Received initial blob with SessionID=0 and non INVITE SLP data
```

Tested with version git in **Debian testing**

PD: When does half run, ie when at least the GUI comes the ability to accept the download (although it does not work hahahaha), get this error:

```
[22:12:38 ERROR papyon.service.content_roaming] Content Roaming service got error: HTTP Error (404): Not Found
[22:12:38 ERROR papyon.service.content_roaming] Content Roaming service got error: HTTP Error (404): Not Found

/usr/share/emesene/emesene/gui/gtkui/FileTransferWidget.py:69: GtkWarning: gtk_box_pack: assertion `child->parent == NULL' failed
  self.pack_start(self.menu)

[22:13:58 WARNING papyon.msnp2p.session_manager] Received initial blob with SessionID=0 and non INVITE SLP data
[22:13:58 WARNING papyon.msnp2p.session_manager] Received initial blob with SessionID=0 and non INVITE SLP data
```

**Comment by c10ud** [comment]
your errors are related to your profile being broken

**Comment by arielj** [comment]
ErunamoJAZZ, can you try with latest git? that last error means that you are not using it (line 69 of FileTranserWidget is not that line on your trace)

**Comment by ErunamoJAZZ** [comment]
arielj... Great!
Not quite perfect (I had to restart a couple of times after some Emesene transfer), but IT WORKS!

But there is a problem with LiveMSN for Mac: https://github.com/emesene/emesene/issues/837
To LiveMSN in Windows, it works.

Thanks :)

**Comment by arielj** [comment]
@danielgomon, can you confirm that it works at least some times?

can you both provide more information from the moments it doesn't work with latest git? (mainly terminal output)

**Comment by danielgomon** [comment]
Today I've cloned the git version again, apparently it works fine now. I've tested several times the file transfer function with a contact who uses emesene 2.11.7 and pidgin. With emesene 2.11.7 I can accept and receive my contact files every time I tried, but sometimes she can't accept my files, that's an emesene 2.11.7 issue, I think. With pidgin the file transfer works ok, I can accept, receive and send files without a problem. 

This is the terminal output, my contact sign-up at first in emesene then she changes to pidgin:

$ python2 ./emesene
*\* Message: pygobject_register_sinkfunc is deprecated (GstObject)
[22:49:09 ERROR papyon.protocol.notification] Notification got error : 913 10
warning: line 6172: incompatible stripping characters and condition
warning: line 6184: incompatible stripping characters and condition
[22:49:31 WARNING papyon.msnp2p.transport] TLPv1 Parse Error: empty chunk for non-empty blob
00 00 00 00 a8 3c 23 10 00 00 00 00 00 00 00 00
e4 04 00 00 00 00 00 00 00 00 00 00 02 00 00 00
24 ac 68 15 06 ee 0e 1b e4 04 00 00 00 00 00 00
[22:50:24 WARNING papyon.msnp2p.transport] TLPv1 Parse Error: empty chunk for non-empty blob
00 00 00 00 aa 3c 23 10 00 00 00 00 00 00 00 00
a5 20 00 00 00 00 00 00 00 00 00 00 00 00 00 00
82 6e 1c 19 00 00 00 00 00 00 00 00 00 00 00 00
[22:50:24 WARNING papyon.msnp2p.transport] TLPv1 Parse Error: empty chunk for non-empty blob
00 00 00 00 ab 3c 23 10 00 00 00 00 00 00 00 00
e4 04 00 00 00 00 00 00 00 00 00 00 02 00 00 00
54 a8 33 4a 55 ee 0e 1b e4 04 00 00 00 00 00 00
[22:50:27 WARNING papyon.msnp2p.transport] TLPv1 Parse Error: chunk end exceeds blob size
00 00 00 00 aa 3c 23 10 2c 1c 00 00 00 00 00 00
a5 20 00 00 00 00 00 00 b2 04 00 00 00 00 00 00
82 6e 1c 19 00 00 00 00 00 00 00 00 00 00 00 00
[22:50:30 WARNING papyon.msnp2p.transport] TLPv1 Parse Error: empty chunk for non-empty blob
00 00 00 00 ad 3c 23 10 00 00 00 00 00 00 00 00
54 01 00 00 00 00 00 00 00 00 00 00 02 00 00 00
f2 de d1 4d d7 b6 4a 5b 54 01 00 00 00 00 00 00
[22:50:37 WARNING papyon.msnp2p.transport] TLPv1 Parse Error: empty chunk for non-empty blob
42 dc b9 13 af 3c 23 10 00 00 00 00 00 00 00 00
b3 3e 02 00 00 00 00 00 00 00 00 00 30 00 00 01
d1 ba ad 2e 00 00 00 00 00 00 00 00 00 00 00 00
[22:50:38 WARNING papyon.msnp2p.transport] TLPv1 Parse Error: empty chunk for non-empty blob
00 00 00 00 b0 3c 23 10 00 00 00 00 00 00 00 00
54 01 00 00 00 00 00 00 00 00 00 00 02 00 00 00
08 23 24 76 dc b6 4a 5b 54 01 00 00 00 00 00 00
[22:50:38 WARNING papyon.msnp2p.transport] TLPv1 Parse Error: empty chunk for non-empty blob
42 dc b9 13 af 3c 23 10 00 00 00 00 00 00 00 00
b3 3e 02 00 00 00 00 00 00 00 00 00 30 00 00 01
d1 ba ad 2e 00 00 00 00 00 00 00 00 00 00 00 00
[22:50:39 WARNING papyon.msnp2p.transport] TLPv1 Parse Error: empty chunk for non-empty blob
00 00 00 00 b1 3c 23 10 00 00 00 00 00 00 00 00
54 01 00 00 00 00 00 00 00 00 00 00 02 00 00 00
a3 26 26 7d de b6 4a 5b 54 01 00 00 00 00 00 00
